### PR TITLE
feat(auth): add updatePassword API

### PR DESCRIPTION
**Pull request by israx**
#### Description of changes

updatePassword API is used to update the user's password while authenticated.

#### Description of how you validated changes

I created a sample app and was able to update the password and log in with changed password.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/auth/__tests__/providers/cognito/updatePassword.test.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { AmplifyErrorString, Amplify } from '@aws-amplify/core';
+import { AuthError } from '../../../src/errors/AuthError';
+import { AuthValidationErrorCode } from '../../../src/errors/types/validation';
+import { updatePassword } from '../../../src/providers/cognito';
+import { ChangePasswordException } from '../../../src/providers/cognito/types/errors';
+import * as changePasswordClient from '../../../src/providers/cognito/utils/clients/ChangePasswordClient';
+import { ChangePasswordCommandOutput } from '@aws-sdk/client-cognito-identity-provider';
```
Comment (jimblanc) [inline packages/auth/__tests__/providers/cognito/updatePassword.test.ts:10]:
```suggestion
import type { ChangePasswordCommandOutput } from '@aws-sdk/client-cognito-identity-provider';
```

File: packages/auth/src/providers/cognito/utils/clients/ChangePasswordClient.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+export async function changePasswordClient(
+	params: ChangePasswordCommandInput
+): Promise<ChangePasswordCommandOutput> {
+	const client = new UserPoolHttpClient(UserPoolClient.region);
+	const result: ChangePasswordCommandOutput =
+		await client.send<ChangePasswordCommandOutput>('ChangePassword', {
+			...params,
+		});
+	return result;
```
Comment (jimblanc) [inline packages/auth/src/providers/cognito/utils/clients/ChangePasswordClient.ts:19]: Nit

```suggestion
	return client.send<ChangePasswordCommandOutput>('ChangePassword', {
		...params,
	});
```

**Review by jimblanc — APPROVED** [review]
Looks good! Few nits

**Review by jimblanc — APPROVED** [review]

**Review by kvramyasri7 — APPROVED** [review]
LGTM

**Review by AllanZhengYP — APPROVED** [review]

**Pull request merged**